### PR TITLE
build: add MSAN config

### PR DESCRIPTION
**Pull request by lizan**

### CODE REVIEW

**Comment by lizan** [comment]
@htuch @mattklein123 ready to go without enable in CI yet, just usable configs and instructions.

File: .bazelrc [hunk 1]
```diff
@@ -84,14 +84,16 @@ build:clang-tsan --linkopt -fuse-ld=lld
...
+# Clang MSAN - this is the base config for remote-msan and docker-msan. To run this config without
+# our build image, follow https://github.com/google/sanitizers/wiki/MemorySanitizerLibcxxHowTo
+# with libc++ instruction and provide corresponding `--copt` and `--linkopt` as well.
 build:clang-msan --action_env=ENVOY_MSAN=1
 build:clang-msan --config=sanitizer
 build:clang-msan --define ENVOY_CONFIG_MSAN=1
 build:clang-msan --copt -fsanitize=memory
 build:clang-msan --linkopt -fsanitize=memory
 build:clang-msan --copt -fsanitize-memory-track-origins=2
+build:clang-msan --copt -O1
```
Comment (mattklein123) [inline .bazelrc]: ?

Comment (lizan) [reply .bazelrc]: https://clang.llvm.org/docs/MemorySanitizer.html#usage

> To get a reasonable performance add -O1 or higher.

Without this, more tests will just timeout.

Comment (mattklein123) [reply .bazelrc]: OK, do you mind adding a comment?

/wait

Comment (lizan) [reply .bazelrc]: done

**Pull request merged**